core/src/main/java/org/prevayler/foundation/ObjectInputStreamWithClassLoader.java

```diff
@@ -18,7 +18,16 @@ public ObjectInputStreamWithClassLoader(InputStream stream, ClassLoader loader)
     _loader = loader;
   }
 
-  protected Class resolveClass(ObjectStreamClass v) throws IOException, ClassNotFoundException {
-    return (_loader != null ? Class.forName(v.getName(), false, _loader) : super.resolveClass(v));
+  protected Class<?> resolveClass(ObjectStreamClass v) throws IOException, ClassNotFoundException {
+    if (_loader == null)
+      return super.resolveClass(v);
+    try {
+      return Class.forName(v.getName(), false, _loader);
+    } catch (ClassNotFoundException ex) {
+      Class<?> cl = super.resolveClass(v); // fall back for resolving primitive types
```

Comment:
This change seems distinct from the generic type cleanup. Did you run into issues with primitive types yourself?

Reply:
Well, not really. But I found several discussions on the Internet regarding problems with primitives in conjunction with custom class loaders. All of them offered the same solution. So it appeared to me as a potential improvement one one hand which on the other hand cannot possibly do any harm to existing software.

Reply:
What is your opinion ? Should we keep the fallback clause or stick with the former version ?

Reply:
Done.